content-api/content-actors/src/main/scala/org/sunbird/content/import/ImportManager.scala

```diff
@@ -0,0 +1,129 @@
+package org.sunbird.content.`import`
+
+import java.util
+import java.util.UUID
+
+import org.apache.commons.collections.CollectionUtils
+import org.apache.commons.collections4.MapUtils
+import org.apache.commons.lang3.StringUtils
+import org.sunbird.common.Platform
+import org.sunbird.common.dto.{Request, Response, ResponseHandler}
+import org.sunbird.common.exception.ClientException
+import org.sunbird.graph.OntologyEngineContext
+import org.sunbird.graph.common.Identifier
+import org.sunbird.telemetry.util.LogTelemetryEventUtil
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+object ImportManager {
+
+  val REQUEST_LIMIT = Platform.getInteger("content.import.request_size_limit", 200)
+  val AUTO_CREATE_TOPIC_NAME = Platform.config.getString("content.import.topic_name")
+  val REQUIRED_PROPS = Platform.getStringList("content.import.required_props", java.util.Arrays.asList("name", "code", "mimeType", "contentType", "artifactUrl", "framework"))
+
+  def importContent(request: Request)(implicit oec: OntologyEngineContext, ec: ExecutionContext): Future[Response] = {
+    val graphId: String = request.getContext.get("graph_id").asInstanceOf[String]
+    val reqList: util.List[util.Map[String, AnyRef]] = getRequest(request)
+    if (CollectionUtils.isNotEmpty(reqList) && reqList.size > REQUEST_LIMIT)
+      throw new ClientException(ImportErrors.ERR_REQUEST_LIMIT_EXCEED, ImportErrors.ERR_REQUEST_LIMIT_EXCEED_MSG + REQUEST_LIMIT)
+    val processId: String = UUID.randomUUID().toString
+    val invalidCodes: util.List[String] = new util.ArrayList[String]()
+    validateAndGetRequest(reqList, processId, invalidCodes).map(list => {
+      if (CollectionUtils.isNotEmpty(invalidCodes)) {
+        val msg = if (invalidCodes.asScala.filter(c => StringUtils.isNotBlank(c)).toList.size > 0) " | Required Property's Missing For " + invalidCodes else ""
+        throw new ClientException(ImportErrors.ERR_REQUIRED_PROPS_VALIDATION, ImportErrors.ERR_REQUIRED_PROPS_VALIDATION_MSG + REQUIRED_PROPS + msg)
+      } else {
+        list.asScala.map(req => {
+          val source: String = req.getOrDefault(ImportConstants.SOURCE, "").toString
+          val identifier = if(StringUtils.isNotBlank(source)) source.substring(source.lastIndexOf('/') + 1) else Identifier.getIdentifier(graphId, Identifier.getUniqueIdFromTimestamp)
```

Comment:
Why are we using this identifier? It should create a new always.

Reply:
As discussed, it is required for bulk approval from dock to diksha.